internal/acceptance_test/resource_host_test.go

```diff
@@ -76,6 +164,24 @@ func testAccCheckHostDestroy(t *testing.T, s *terraform.State) error {
 		if rs.Type != "test_host" {
 			continue
 		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No host primary ID set")
+		}
+
+		hostID := rs.Primary.ID
+
+		p, err := client.GetClientFromMetaMap(testAccProvider.Meta())
+		if err != nil {
+			return fmt.Errorf("Error retrieving Metal client: %v", err)
+		}
+
+		ctx := p.GetContext()
+
+		host, _, err := p.Client.HostsApi.GetByID(ctx, hostID)
+		if err == nil && host.State != rest.HOSTSTATE_DELETED {
+			return fmt.Errorf("Host: %s still exists", hostID)
```

Comment:
What if GetByID returns error?  That's counted as Host deleted?  Might be better to check for specific err of not found if possible.  Otherwise, any error could be considered success/deleted when the host might still be there.